### ISSUE TITLE
roadmap: add reproducible post-training ladder (SFT → preference optimization → RL-style training → distributed scale)

### ISSUE THREAD

**Issue by rmems** [rmems/agoge-forger#75]
## Goal
Close the current post-training / large-scale training experience gap by turning `agoge-forger` into a reproducible post-training research platform rather than only a QLoRA fine-tuning utility.

## Scope
Create a staged post-training roadmap and implementation plan covering:

- M0: supervised fine-tuning baseline on a small open model
- M1: held-out coding / reasoning evals
- M2: preference dataset ingestion from `operation-prometheus`
- M3: one preference-optimization method (DPO first unless evidence supports another)
- M4: verifier / reward layer
- M5: one RL-style training method
- M6: single-node multi-GPU execution
- M7: cloud distributed run
- M8: reproducible report with base-vs-post-trained comparisons

## Requirements
- Preserve current PyTorch-first architecture.
- Keep configs declarative and reproducible.
- Record exact dataset version, seed, hyperparameters, environment, GPU topology, wall-clock time, throughput, checkpoints, and final artifacts in manifests.
- Do not add multiple algorithms at once; land one validated path per milestone.
- Local RTX 5080 runs must remain first-class smoke / development paths.
- Cloud scale should reuse the same recipe rather than creating a separate ad hoc trainer.

## Deliverables
- `docs/post_training_roadmap.md`
- milestone checklist with issue links
- explicit acceptance criteria for each stage
- architecture note describing the contract with `operation-prometheus`

## Acceptance criteria
- Roadmap is committed and references concrete follow-up issues.
- Each milestone defines a reproducible command, artifact outputs, and evaluation gate.
- No claim of "frontier-scale" training is made until a real distributed run and failure-recovery evidence exist.

## Current critical path (2026-08-30)

The active local SFT path is now:

```text
#112 repository-boundary and metadata cleanup
→ #99 immutable data/split contract
→ #100 held-out base-vs-SFT evaluation
→ #104 compatibility bakeoff
→ #101 measured Granite 4.1 pilot
→ #102 Hugging Face release
→ #103 multi-source ablation
```

#77–#81 remain the later post-training/distributed ladder and must not preempt the first published SFT result. Repository ownership outside Agoge is tracked by linked producer/infrastructure issues rather than duplicated implementation trees.

**Comment by rmems** [comment]
## Ownership note (2026-08-12)

Post-training ladder (**this issue and #76–#81**) stays in **agoge-forger**.

**GPU kernels** (L1–L3 on the RTX 5080 host) are SoT in [`rmems/blackwell-kernel-lab`](https://github.com/rmems/blackwell-kernel-lab) — see epic [#1](https://github.com/rmems/blackwell-kernel-lab/issues/1) and `docs/KERNELS.md` there.

- Multi-GPU / multi-node issues (#80/#81) = **distributed training experience** (PyTorch), not custom CUDA kernel ownership.
- Do not grow `cuda/` in this repo into a second kernel tree.

**Comment by rmems** [comment]
**DELETE** — 04 Agoge Technical Debt Reducer

Debt-mode KEEP/DELETE triage for `rmems/agoge-forger#75`.
This is a training/experiment/roadmap item. Agoge maintenance mode forbids new training mills, model downloads, eval campaigns, and cloud jobs. Recommend close unless Raul reopens with a named train brief.

**Comment by rmems** [comment]
Closed per Tech Debt Mill DELETE triage (debt mode: not planned / frozen train-experiment-research). Reopen only with a named brief.
